demisto_sdk/commands/doc_reviewer/doc_reviewer.py

```diff
@@ -0,0 +1,372 @@
+                            f"{file}:", fg='bright_red')
+                self.print_unknown_words()
+                self.found_misspelled = True
+                self.files_with_misspells.add(file)
+
+            else:
+                click.secho(f" - No misspelled words found in {file}", fg='green')
+                self.files_without_misspells.add(file)
+
+        self.print_file_report()
+        if self.found_misspelled and not self.no_failure:
+            return False
+
+        return True
+
+    def add_known_words(self):
+        """Add known words to the spellchecker from external and internal files"""
+        # adding known words file if given - these words will not count as misspelled
+        if self.known_words_file_path:
+            self.spellchecker.word_frequency.load_text_file(self.known_words_file_path)
+
+        # adding the KNOWN_WORDS to the spellchecker recognized words.
+        self.spellchecker.word_frequency.load_words(KNOWN_WORDS)
+
+        if self.expand_dictionary:
+            # nltk - natural language tool kit - is a large package containing several dictionaries.
+            # to use it we need to download one of it's dictionaries - we will use the
+            # reasonably sized "brown" and "webtext" dicts.
+            # to avoid SSL download error  we disable SSL connection.
+            try:
+                _create_unverified_https_context = ssl._create_unverified_context
+            except AttributeError:
+                pass
+            else:
+                ssl._create_default_https_context = _create_unverified_https_context
+
+            # downloading "brown" and "webtext" sets from nltk.
+            click.secho("Downloading expanded dictionary, this may take a minute...", fg='yellow')
+            nltk.download('brown')
+            nltk.download('webtext')
+
+            # adding nltk's word set to spellchecker.
+            self.spellchecker.word_frequency.load_words(brown.words())
+            self.spellchecker.word_frequency.load_words(webtext.words())
+
+    @staticmethod
+    def remove_punctuation(word):
+        """remove leading and trailing punctuation"""
+        return word.strip(string.punctuation)
+
+    def check_word(self, word):
+        """Check if a word is legal"""
+        # check camel cases
+        if not self.no_camel_case and self.is_camel_case(word):
+            sub_words = self.camel_case_split(word)
+            for sub_word in sub_words:
+                sub_word = self.remove_punctuation(sub_word)
+                if sub_word.isalpha() and self.spellchecker.unknown([sub_word]):
+                    self.unknown_words[word] = list(self.spellchecker.candidates(sub_word))[:5]
+
+        else:
+            word = self.remove_punctuation(word)
+            if word.isalpha() and self.spellchecker.unknown([word]):
+                self.unknown_words[word] = list(self.spellchecker.candidates(word))[:5]
+
+    def check_md_file(self, file_path):
```

Comment:
move this to the relevant content objects
currently, you open each file multiple times.